plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/internal/CloudSdkUpdateJob.java

```diff
@@ -51,7 +52,7 @@ public CloudSdkUpdateJob(MessageConsoleStream consoleStream, ReadWriteLock cloud
    */
   @Override
   protected IStatus modifySdk(IProgressMonitor monitor) {
-    // todo should we convert to a submonitor here?
+    monitor = SubMonitor.convert(monitor, 30);
     
     monitor.beginTask(Messages.getString("configuring.cloud.sdk"), 10); //$NON-NLS-1$
```

Comment:
These two lines should be merged.  And looking further down, the calls where we use the ProgressWrapper should then use `new ProgressWrapper(monitor.split(XXX))`.  And the `ProgressWrapper` constructor should take a `SubMonitor`.

Reply:
Why should these two lines be merged?

Reply:
More efficient, and `SubMonitor.convert(IPM)` calls `SubMonitor.convert(IPM, "", BIGNUMBER)` under the hood.